lib/ASRouter.jsm

```diff
@@ -55,21 +55,45 @@ const MessageLoaderUtils = {
    *
    * @param {obj} provider An AS router provider
    * @param {string} provider.url An endpoint that returns an array of messages as JSON
+   * @param {obj} storage A storage object with get() and set() methods for caching.
    * @returns {Promise} resolves with an array of messages, or an empty array if none could be fetched
    */
-  async _remoteLoader(provider) {
+  async _remoteLoader(provider, storage) {
```

Comment:
You need to add a special case never to cache preview URLs somehow – maybe check if `updateCycleInMs` is 0?

Reply:
Awesome, works great!